editor/editor_node.cpp

```diff
@@ -1347,7 +1347,17 @@ void EditorNode::_find_node_types(Node *p_node, int &count_2d, int &count_3d) {
 void EditorNode::_save_scene_with_preview(String p_file, int p_idx) {
 	EditorProgress save("save", TTR("Saving Scene"), 4);
 
+	Ref<World> edited_world;
+
 	if (editor_data.get_edited_scene_root() != nullptr) {
+		// Allow a generic mechanism for the engine to make changes prior, and after saving.
+		if (editor_data.get_edited_scene_root()->get_tree() && editor_data.get_edited_scene_root()->get_tree()->get_root()) {
+			edited_world = editor_data.get_edited_scene_root()->get_tree()->get_root()->get_world();
+			if (edited_world.is_valid()) {
+				edited_world->notify_saving(true);
+			}
+		}
```

Comment:
This hook could be really useful to have in editor plugins and `@tool` scripts, also in 4.x :slightly_smiling_face: 

There are multiple instances where I've wanted to make editor-only changes that aren't serialized to scene files outside of LOD.

Reply:
Yes I was surprised there wasn't something like this already. Although it may have to be done carefully in some cases to avoid race conditions.